### PR TITLE
An initial approach of frontend arch

### PR DESCRIPTION
**Pull request by rafaparadela**
This request implements several functionalities in the client side. It only adds an initial  approach, trying to keep separated side effect functions.

@dialelo Could you review please? Thank you!!

### CODE REVIEW

File: scala-exercises-client/src/main/scala/scripts/ExercisesJS.scala [hunk 1]
```diff
@@ -1,17 +1,55 @@
...
+
+    def updateExerciseList(ex: List[ClientExercise]): IO[Unit] = io(exercises() = ex)
+
+    case class ClientExercise(method: String, arguments: Seq[String] = Nil) {
+      def isFilled: Boolean = !arguments.exists(_.isEmpty) && arguments.nonEmpty
+    }
+
+    val exercises = Var(getMethodsList map (m ⇒ ClientExercise(m)) toList)
+
+    exercises foreach (e ⇒ println(e))
```
Comment (raulraja) [inline scala-exercises-client/src/main/scala/scripts/ExercisesJS.scala]: This should be removed before merging

File: scala-exercises-client/src/main/scala/utils/DomHandler.scala [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+object DomHandler {
+
+  import IO._
+
+  def replaceInputs(nodes: Seq[(HTMLElement, String)]): IO[Unit] = io {
+    nodes foreach { case (n, r) ⇒ $(n).html(r) }
+  }
+
+  def onInputKeyUp(onkeyup: (String, Seq[String]) ⇒ IO[Unit]): IO[Unit] = io {
```
Comment (raulraja) [inline scala-exercises-client/src/main/scala/utils/DomHandler.scala]: All the relevant public methods should include scaladoc

File: scala-exercises-client/src/main/scala/scripts/ExercisesJS.scala [hunk 3]
```diff
@@ -1,17 +1,55 @@
...
-  def runHighlight = $("pre code").each({ (_: Any, code: dom.Element) ⇒
-    js.Dynamic.global.hljs.highlightBlock(code)
-  })
+  import Model._
+
+  def inputChanged(method: String, args: Seq[String]): IO[Unit] =
+    updateExerciseList(updateExcercise(method, args))
+
+  def inputBlur(method: String): IO[Unit] = io {
+    println("Holaaaa")
```
Comment (dialelo) [inline scala-exercises-client/src/main/scala/scripts/ExercisesJS.scala]: We should remove this println :fire:

File: scala-exercises-client/src/main/scala/scripts/ExercisesJS.scala [hunk 4]
```diff
@@ -1,17 +1,55 @@
...
+    Model.getExercise(method) foreach (e ⇒ println(e.isFilled))
+  }
+
+  def main(): Unit = {
+
+    val program = for {
+      _ ← replaceInputs(insertInputs)
+      //      _ ← onInputKeyUp(inputChanged)
```
Comment (dialelo) [inline scala-exercises-client/src/main/scala/scripts/ExercisesJS.scala]: We can delete this line and add it later.

**Comment by dialelo** [comment]
LGTM :shipit:

**Pull request merged**